frontend/src/components/general/LoginNudge.tsx

```diff
@@ -23,28 +25,32 @@ type Props = {
   fullPage?: boolean;
   className?: string;
 };
+
 export default function LoginNudge({ whatToDo, fullPage, className }: Props) {
   const classes = useStyles();
   const { locale } = useContext(UserContext);
   const texts = getTexts({ page: "general", locale: locale });
-  const urlParams = new URLSearchParams(window.location.search);
-  const hub = urlParams.get("hub");
+  const router = useRouter();
+  const currentPath = router.asPath;
+  const encodedRedirectUrl = encodeURIComponent(currentPath);
+  const hubUrl = extractHubFromUrl(currentPath);
```

Comment:
either, we can be sure, that we can get the hub via: 
```
const urlParams = new URLSearchParams(window.location.search);
const hub = urlParams.get("hub");
```
or the parent component should be dealing with that, shouldn't it? It might know it's hub property anyways.